### PR TITLE
DT-2229 move queue properties to library

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: test-app/helm_deploy/hmpps-template-kotlin/values.yaml [hunk 1]
```diff
@@ -30,6 +30,22 @@ generic-service:
...
+      HMPPS_SQS_QUEUES_MAINQUEUE_DLQ_SECRET_ACCESS_KEY: "dlq_secret_access_key"
+      HMPPS_SQS_QUEUES_MAINQUEUE_DLQ_NAME: "mainDlq"
+    another-queue-sqs-instance-output:
+      HMPPS_SQS_QUEUES_ANOTHERQUEUE_ACCESS_KEY_ID: "another_access_key_id"
+      HMPPS_SQS_QUEUES_ANOTHERQUEUE_SECRET_ACCESS_KEY: "another_secret_access_key"
+      HMPPS_SQS_QUEUES_ANOTHERQUEUE_QUEUE_NAME: "anotherQueue"
+    another-queue-sqs-dl-instance-output:
+      HMPPS_SQS_QUEUES_ANOTHERQUEUE_DLQ_ACCESS_KEY_ID: "dlq_another_access_key_id"
+      HMPPS_SQS_QUEUES_ANOTHERQUEUE_DLQ_SECRET_ACCESS_KEY: "dlq_another_secret_access_key"
+      HMPPS_SQS_QUEUES_ANOTHERQUEUE_DLQ_NAME: "anotherDlq"
```
Comment (mikehalmamoj) [inline test-app/helm_deploy/hmpps-template-kotlin/values.yaml:48]: Add examples of how to wire up Kube secrets into HmppsQueueProperties

File: test-app/src/main/resources/application.yml [hunk 2]
```diff
@@ -60,9 +60,18 @@ oauth:
...
+      dlqName: aws-dlq
+      dlqAccessKeyId: dlq-access-key-id
+      dlqSecretAccessKey: dlq-secret-access-key
+    anotherQueue:
+      queueName: another-aws-queue
+      queueAccessKeyId: another-queue-access-key-id
+      queueSecretAccessKey: another-queue-secret-access-key
+      dlqName: another-aws-dlq
+      dlqAccessKeyId: another-dlq-access-key-id
+      dlqSecretAccessKey: another-dlq-secret-access-key
```
Comment (mikehalmamoj) [inline test-app/src/main/resources/application.yml:77]: Added a 2nd queue to the functional tests to show we can handle multiple

File: test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/HmppsQueueResourceTest.kt [hunk 3]
```diff
@@ -76,16 +77,36 @@ class HmppsQueueResourceTest : IntegrationTestBase() {
...
   @Nested
   inner class RetryAllDlqs {
-    // TODO when we start handling multiple queues add another queue/dlq combination to this test
     @Test
     fun `should transfer messages from DLQ to main queue and process them`() {
       sqsDlqClient.sendMessage(dlqUrl, "message1")
-      sqsDlqClient.sendMessage(dlqUrl, "message2")
-      await untilCallTo { sqsDlqClient.countMessagesOnQueue(dlqUrl) } matches { it == 2 }
+      anotherSqsDlqClient.sendMessage(anotherDlqUrl, "message2")
+      await untilCallTo { sqsDlqClient.countMessagesOnQueue(dlqUrl) } matches { it == 1 }
```
Comment (mikehalmamoj) [inline test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/HmppsQueueResourceTest.kt:108]: Now adds a message to each dlq to prove that all dlqs are processed by the `/retry-dlqs` endpoint.

File: test-app/helm_deploy/hmpps-template-kotlin/values.yaml (hunk 1 above)
Comment (mikehalmamoj) [reply test-app/helm_deploy/hmpps-template-kotlin/values.yaml:48]: Oh and tried to write a test to prove that this format of env vars is wored into the HmppsQueueProperties but couldn't manage it - but did manage to test from the command line.

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**